plugin/forward/persistent.go

```diff
@@ -17,9 +16,9 @@ type persistConn struct {
 
 // Transport hold the persistent cache.
 type Transport struct {
-	avgDialTime int64                     // kind of average time of dial time
-	conns       map[string][]*persistConn // Buckets for udp, tcp and tcp-tls.
-	expire      time.Duration             // After this duration a connection is expired.
+	avgDialTime int64             // kind of average time of dial time
+	conns       [3][]*persistConn // Buckets for udp, tcp and tcp-tls.
```

Comment:
don't you want to define a constant like `typeTotal` and put it last in the list of `typeUdp`, `typeUdp`, `typeUdp`? 
Then you could specify array size as
```
    conns       [typeTotal][]*persistConn 
```
without any magic numbers

Also, in future, you will be able to add one more transport type without changing array definition/initialization code.

Reply:
yep good idea.